server/src/main/java/org/opensearch/search/profile/query/InternalProfileCollectorManager.java

```diff
@@ -46,9 +48,33 @@ public InternalProfileCollectorManager(
     ) {
         this.manager = manager;
         this.reason = reason;
+        this.collectorManagerName = deriveCollectorManagerName(manager);
         this.children = children;
     }
 
+    /**
+     * Creates a human-friendly representation of the CollectorManager name.
+     *
+     * @param manager The CollectorManager to derive a name from
+     * @return  A (hopefully) prettier name
+     */
+    private String deriveCollectorManagerName(CollectorManager<? extends Collector, ReduceableSearchResult> manager) {
+        String name = manager.getClass().getSimpleName();
+        if (name.equals("")) {
+            name = manager.getClass().getEnclosingClass().getSimpleName();
+        }
+
+        // Include the user-defined agg name
+        String userDefinedAggName = null;
+        if (manager instanceof AggregationCollectorManager) {
+            userDefinedAggName = ((AggregationCollectorManager) manager).getCollectorName();
+        }
+        if (userDefinedAggName != null) {
+            name += ": [" + userDefinedAggName + "]";
```

Comment:
we don't need userDefinedAggName parameter here. Instead you can do below:

```
if (manager instanceof AggregationCollectorManager) {
   name += ": [" + ((AggregationCollectorManager) manager).getCollectorName() + "]";
}
```

Reply:
It has been addressed now. Thanks